### DISCUSSION TITLE
How to change docker image names on deployment with azd

### DISCUSSION THREAD

**Discussion by christiannagel** [microsoft/aspire#2165] [Q&A]
How can I customize a docker image name? 

Using `azd up`, the images are named **azd-deploy-container-identifier** with the latest tag. The identifier changes with every new deployment. 
I didn't find information about how to customize this. It looks like the element `ContainerRepository` from the csproj file isn't used.

**Comment by davidfowl** [comment]
AFAIK azd is using container publish https://learn.microsoft.com/en-us/dotnet/core/docker/publish-as-container?pivots=dotnet-8-0. Defaulting the container name to the resource name. I think you can override it.

cc@baronfel

**Comment by christiannagel** [comment]
Thanks @davidfowl 

@baronfel, `azd deploy` creates a new repository every time this command runs, e.g.
azd-deploy-bot-1707570986
azd-deploy-bot-1707571578
all with the latest tag.

I would like to override the name, e.g. codebreaker-bot, use tags for the version, and the latest tag for the newest build.
`azd deploy` might use `dotnet publish` but doesn't use the csproj setting for `ContainerRepository`. 

How can this be customized?

**Reply by baronfel** [reply]
I believe azd is overriding the image name via that property at the command line. Command line properties are 'global properties' in MSBuild terminology, and cannot be changed by your project file.

I noticed while pairing with @davidfowl earlier this week that azd always generates these image names, but never uses a base name with a tag. We should figure out why that is, and if we can get them to use ContainerImageTag(s) for this instead of forcibly overwriting a users specified ContainerRepository.

**Comment by davidfowl** [comment]
@christiannagel can you file an issue on https://github.com/Azure/azure-dev/issues?

**Reply by christiannagel** [reply]
@davidfowl - done: https://github.com/Azure/azure-dev/issues/3335
